pkg/ebpf/driver_interface.go

```diff
@@ -0,0 +1,222 @@
+// +build windows
+
+package ebpf
+
+/*
+//! Defines the objects used to communicate with the driver as well as its control codes
+#include "c/ddfilterapi.h"
+
+//! These includes are needed to use constants defined in the ddfilterapi
+#include <WinDef.h>
+#include <WinIoCtl.h>
+*/
+import "C"
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"unsafe"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"golang.org/x/sys/windows"
+)
+
+const (
+	driverFile = `\\.\ddfilter`
+)
+
+var (
+	// Buffer holding datadog driver filterapi (ddfilterapi) signature to ensure consistency with driver.
+	ddAPIVersionBuf = makeDDAPIVersionBuffer(C.DD_FILTER_SIGNATURE)
+)
+
+// Creates a buffer that Driver will use to verify proper versions are communicating
+// We create a buffer because the system calls we make need a *byte which is not
+// possible with const value
+func makeDDAPIVersionBuffer(signature uint64) []byte {
+	buf := make([]byte, C.sizeof_uint64_t)
+	binary.LittleEndian.PutUint64(buf, signature)
+	return buf
+}
+
+// DriverInterface holds all necessary information for interacting with the windows driver
+type DriverInterface struct {
+	driverHandle windows.Handle
+	iocp         windows.Handle
+	path         string
+}
+
+// NewDriverInterface returns a DriverInterface struct for interacting with the driver
+func NewDriverInterface() (*DriverInterface, error) {
+	dc := &DriverInterface{
+		path: driverFile,
+	}
+
+	handle, err := dc.NewDriverHandle()
+	if err != nil {
+		return nil, fmt.Errorf("%s : %s", "error creating driver handle", err)
+	}
+
+	// Create IO Completion port that we'll use to communicate with the driver
+	iocp, err := windows.CreateIoCompletionPort(handle, windows.Handle(0), 0, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create IO completion port %v", err)
+	}
+
+	dc.driverHandle = handle
+	dc.iocp = iocp
+
+	// Set the packet filters that will determine what we pull from the driver
+	// TODO: Determine failure condition for not setting filter
+	// TODO: I.e., one or more, all fail? I.E., at what point do we not create a tracer
```

Comment:
I think we can remove this comment for now. There's only going to be one filter for the time being, so we can do the easy thing and fail to create the tracer if this happens.